src/Domain/CoordinateMaps/FocallyLiftedMap.cpp

```diff
@@ -0,0 +1,342 @@
+  const return_type lambda_tilde = 1.0 / (1.0 - sigma * (1.0 - lambda));
+
+  // Derivative of lambda_tilde
+  const auto d_lambda_tilde_d_mapped_coords =
+      inner_map_.deriv_lambda_tilde(lower_coords, lambda_tilde, proj_center_);
+
+  // Deriv of x_0 with respect to x
+  auto dx0_dx =
+      make_with_value<tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame>>(
+          dereference_wrapper(source_coords[0]), 0.0);
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t j = 0; j < 3; ++j) {
+      dx0_dx.get(i, j) = gsl::at(d_lambda_tilde_d_mapped_coords, j) *
+                         (gsl::at(lower_coords, i) - gsl::at(proj_center_, i)) /
+                         lambda_tilde;
+    }
+    dx0_dx.get(i, i) += lambda_tilde;
+  }
+
+  // Deriv of sigma with respect to x,y,z
+  auto d_sigma_d_mapped_coords =
+      make_with_value<tnsr::i<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame>>(
+          sigma, 0.0);
+  for (size_t i = 0; i < 3; ++i) {
+    auto tmp = make_with_value<return_type>(lambda, 0.0);
+    for (size_t j = 0; j < 3; ++j) {
+      tmp += gsl::at(d_lambda_d_lower_coords, j) * dx0_dx.get(j, i);
+    }
+    d_sigma_d_mapped_coords.get(i) =
+        (sigma * tmp +
+         gsl::at(d_lambda_tilde_d_mapped_coords, i) / square(lambda_tilde)) /
+        (1.0 - lambda);
+  }
+
+  const auto dxbar_dx_inner = inner_map_.inv_jacobian(source_coords);
+  const auto dxbar_dsigma = inner_map_.dxbar_dsigma(source_coords);
+
+  auto inv_jacobian_matrix =
+      make_with_value<tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame>>(
+          dereference_wrapper(source_coords[0]), 0.0);
+
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t j = 0; j < 3; ++j) {
+      for (size_t k = 0; k < 3; ++k) {
+        inv_jacobian_matrix.get(i, j) +=
+            dxbar_dx_inner.get(i, k) * dx0_dx.get(k, j);
+      }
+      inv_jacobian_matrix.get(i, j) +=
+          gsl::at(dxbar_dsigma, i) * d_sigma_d_mapped_coords.get(j);
+    }
+  }
+
+  return inv_jacobian_matrix;
+}
+
+template <typename InnerMap>
+boost::optional<std::array<double, 3>> FocallyLiftedMap<InnerMap>::inverse(
+    const std::array<double, 3>& target_coords) const noexcept {
+
+  // Scale factor taking target_coords to lower_coords.
+  const auto lambda_tilde =
+      inner_map_.lambda_tilde(target_coords, proj_center_);
+
+  // Cannot find scale factor, so we are out of range of the map.
+  if (not lambda_tilde) {
+    return boost::none;
+  }
+
+  // Try to find lambda_bar going from target_coords to sphere.
+  const auto lambda_bar = FocallyLiftedMapHelpers::try_scale_factor(
+      target_coords, proj_center_, center_, radius_,
+      not InnerMap::projection_source_is_between_focus_and_target(),
+      InnerMap::projection_source_is_between_focus_and_target());
+
+  // Cannot find scale factor, so we are out of range of the map.
+  if (not lambda_bar) {
+    return boost::none;
+  }
+
+  // compute sigma in a roundoff-friendly way.
```

Comment:
As far as I can tell these expressions have pretty much the same precision properties, but this is fairly hard to analyze and I'm not confident in that result.  (Numerically, it looks like they both have error ~ `eps / max(|lambda_tilde - 1|, |lambda_bar - 1|)`.)  It may depend on exactly how the input variables are calculated.

Reply:
I think it does depend on how input variables are calculated.  In early testing, occasionally the random number tests failed with roundoff before I did this.